core/src/main/java/org/elasticsearch/action/support/replication/ReplicationResponse.java

```diff
@@ -171,7 +177,7 @@ public void writeTo(StreamOutput out) throws IOException {
 
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-            builder.startObject(_SHARDS);
+            builder.startObject();
             builder.field(TOTAL, total);
```

Comment:
It has been moved to `DocWriteResponse.innerToXContent()` where 
```
shardInfo.toXContent(builder, params);
```
has been changed to:

```
builder.field(_SHARDS, shardInfo);
```

so that `ShardInfo` is now a `ToXContentObject`. 

This is important for the `ObjectParser` used to parse the `IndexResponse` which can then directly use `ShardInfo.fromXContent(p)`.